lib/Listener/ExifMetadataProvider.php

```diff
@@ -122,8 +126,19 @@ private function gpsDegreesToDecimal($coordinates, ?string $hemisphere): float {
 		$sign = ($hemisphere === 'W' || $hemisphere === 'S') ? -1 : 1;
 		return $sign * ($degrees + $minutes / 60 + $seconds / 3600);
 	}
-
-	private function parseGPSData(string $rawData): float {
+	
+	/**
+	 * @param array|string $rawData
+	 */
+	private function parseGPSData($rawData): float {
+		if (is_array($rawData)) {
+			if (count($rawData) === 3) {
+				return floatval($rawData[0]) + floatval($rawData[1]) / 60 + floatval($rawData[2]) / 3600;
```

Comment:
Why are we doing that computation here? It looks like the coordinate computation that we are doing above.
Could you add some comment explaining the different formats?

Reply:
After reviewing the correct implementation of EXIF metadata for altitude you are correct, there is no need for this.  Apparently some cameras/applications add invalid metadata.  Altitude in only parsed with this parseGPSData and I was trying to correct for that.  I believe just adding a string check to line 98 may be the better solution?

`if (array_key_exists('GPSAltitude', $rawExifData['GPS']) && array_key_exists('GPSAltitudeRef', $rawExifData['GPS']) && is_string($rawExifData['GPS']['GPSAltitude'])) {`

Reply:
If this only occur with `GPSAltitude`, then yes. But I am wondering if we could handle the array format. Maybe add the `if` condition to prevent the error, but also log the value so we can have more information to improve the handling in the future?

Reply:
#2541 Has an image that has Altitude as '0 0 0' and breaks without checking if it is a string.  The GPS Data also appears to have been scrubbed from the image instead of just removing it with latitude and longitude also both being 0.  From reading some exif documentation it should just be a rational64u formatted as numerator/denominator.  If the data isn't valid is there really anything else to do with it?  It doesn't appear that Photos is actually doing anything with the altitude value either.